File: app/src/main/java/com/example/fruitsdiary/usecase/addeditentry/AddEditEntryContract.java
package com.example.fruitsdiary.usecase.addeditentry;

import com.example.fruitsdiary.BasePresenter;
import com.example.fruitsdiary.BaseView;
import com.example.fruitsdiary.model.Entry;
import com.example.fruitsdiary.model.FruitEntry;

import java.util.List;

interface AddEditEntryContract {

    interface View extends BaseView {
        void updateEntryView(Entry entry);
        void onEntrySaved(List<FruitEntry> fruitEntryList);
        void onEntryDeleted();
    }

    interface Presenter extends BasePresenter<View> {
        void saveEntry();
        void deleteEntry();
        void getEntry(int id);
    }
}

File: app/src/main/java/com/example/fruitsdiary/usecase/addeditentry/AddEditEntryContract.kt
package com.example.fruitsdiary.usecase.addeditentry

import com.example.fruitsdiary.BasePresenter
import com.example.fruitsdiary.BaseView
import com.example.fruitsdiary.model.Entry
import com.example.fruitsdiary.model.FruitEntry

interface AddEditEntryContract {

    interface View : BaseView {
        fun updateEntryView(entry: Entry)
        fun onEntrySaved(fruitEntryList: List<FruitEntry>)
        fun onEntryDeleted()
    }

    interface Presenter : BasePresenter<View> {
        fun saveEntry()
        fun deleteEntry()
        fun getEntry(id: Int)
    }
}
